src/Appwrite/Platform/Tasks/ScheduleExecutions.php

```diff
@@ -52,42 +50,54 @@ protected function enqueueResources(Group $pools, Database $dbForPlatform, calla
             }
 
             $scheduledAt = new \DateTime($schedule['schedule']);
-            if ($scheduledAt <= $intervalEnd) {
-                continue;
-            }
 
             $data = $dbForPlatform->getDocument(
                 'schedules',
                 $schedule['$id'],
             )->getAttribute('data', []);
 
             $delay = $scheduledAt->getTimestamp() - (new \DateTime())->getTimestamp();
+            $delay = max($delay, 0);
 
             $this->updateProjectAccess($schedule['project'], $dbForPlatform);
 
-            \go(function () use ($queueForFunctions, $schedule, $scheduledAt, $delay, $data) {
+            \go(function () use ($queueForFunctions, $schedule, $scheduledAt, $delay, $data, $dbForPlatform) {
                 Co::sleep($delay);
 
-                $queueForFunctions->setType('schedule')
+                $executedAt = new \DateTime();
+                $actualDelay = $executedAt->getTimestamp() - $scheduledAt->getTimestamp();
+
+                // Add headers for delayed executions
+                $headers = $data['headers'] ?? [];
+                if ($actualDelay > 0) {
```

Comment:
Let's remove this `if`, let's always have the new headers

Reply:
Let's also ensure with no delay, scheduled-at and executed-at headers are same, and delay is 0.

For this, let's also update test. test function code would log values of these 3 headers, and in test we then look into execution logs, and ensure expected values